operators/src/source/gdal_source.rs

```diff
@@ -168,59 +192,84 @@ impl TimeIntervalInformation {
     }
 }
 
-/// A provider of tile (size) information for a raster/grid
 #[derive(Debug, Serialize, Deserialize, Clone, Copy)]
 pub struct TilingInformation {
-    pub global_pixel_size: Dim2D,
-    pub tile_pixel_size: Dim2D,
+    pub x_axis_tiles: usize,
+    pub y_axis_tiles: usize,
+    pub x_axis_tile_size: usize,
+    pub y_axis_tile_size: usize,
+}
+
+/// A provider of tile (size) information for a raster/grid
+#[derive(Debug, Serialize, Deserialize, Clone, Copy)]
+pub struct TilingStrategy {
+    pub bounding_box: BoundingBox2D,
+    pub tile_pixel_size: ArrayShape2D,
     pub geo_transform: GeoTransform,
 }
 
-impl TilingInformation {
-    fn upper_left_coordinate(&self) -> Coordinate2D {
-        self.geo_transform.upper_left_coordinate
+impl TilingStrategy {
+    pub fn upper_left_pixel_idx(&self) -> GridIdx2D {
+        self.geo_transform
+            .coordinate_to_grid_idx_2d(self.bounding_box.upper_left())
+    }
+
+    pub fn lower_right_pixel_idx(&self) -> GridIdx2D {
+        let lr_idx = self
+            .geo_transform
+            .coordinate_to_grid_idx_2d(self.bounding_box.lower_right());
+
+        lr_idx - 1
     }
 
-    fn lower_right_coordinate(&self) -> Coordinate2D {
-        self.geo_transform
-            .grid_2d_to_coordinate_2d(self.global_pixel_size.as_pattern())
+    pub fn pixel_idx_to_tile_idx(&self, pixel_idx: GridIdx2D) -> GridIdx2D {
```

Comment:
here I think having two distinct types for pixel index and tile index would be beneficial (newtype/alias would suffice)

Reply:
while this is possible i think it will generate more confusion since you will have to unpack them to use add and so on :( So i vote against this